### PR TITLE
[#2460] Disable preview button if you have unsaved changes. This way you...

### PR DESCRIPTION
**Pull request by ScottDowne**
... cannot preview. One disabled state for buttons. Changing title opens share and saves.

### CODE REVIEW

File: css/buttons.less [hunk 1]
```diff
@@ -37,6 +38,12 @@
   }
 }
 
+
+.butter-btn.butter-disabled {
+  opacity: 0.5;
+  cursor: not-allowed;
+}
```
Comment (ScottDowne) [inline css/buttons.less:44]: One disabled state to rule them all.

File: css/buttons.less [hunk 2]
```diff
@@ -61,12 +68,6 @@
   &:hover {
     color: #FFF;
   }
-
-  &.disabled {
-    color: #EEE;
-    background: @greyGreen;
-    border: none;
-  }
```
Comment (ScottDowne) [inline css/buttons.less:69]: Removing second disabled state, now just have one for all buttons to use.

File: src/editor/share-editor.js [hunk 3]
```diff
@@ -100,10 +115,7 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
 
       projectName.value = project.name;
       projectURL.value = project.publishUrl;
-      previewBtn.href = project.publishUrl;
-
-      headerPreviewBtn.classList.remove( "butter-hidden" );
-      headerPreviewBtn.href = project.publishUrl;
```
Comment (ScottDowne) [inline src/editor/share-editor.js:106]: Removing all references to header preview inside share editor. Header  code handles those buttons now.

File: src/ui/header.js [hunk 4]
```diff
@@ -139,6 +151,7 @@ define([ "dialog/dialog", "util/lang", "ui/user-data", "ui/widget/tooltip" ],
       _projectName.textContent = node.value || _projectTitlePlaceHolderText;
       if( checkProjectName( _projectName.textContent ) ) {
         butter.project.name = _projectName.textContent;
+        login( prepare );
```
Comment (ScottDowne) [inline src/ui/header.js:154]: This makes sure our share editor is opened after you change the title in the header, so you do not get confused when you open the share editor and it is asking for the title again.

File: src/editor/share-editor.js [hunk 5]
```diff
@@ -79,6 +79,22 @@ define([ "editor/editor", "editor/base-editor", "ui/user-data",
       " frameborder='0' mozallowfullscreen webkitallowfullscreen allowfullscreen></iframe>";
     }
 
+    function togglePreviewButton( on ) {
+      if ( on ) {
+        previewBtn.classList.remove( "butter-disabled" );
+        previewBtn.href = butter.project.publishUrl;
+        previewBtn.onclick = function() {
+          return true;
```
Comment (cadecairos) [inline src/editor/share-editor.js:87]: is this line necessary? includes the `return false` below too

File: src/ui/header.js [hunk 6]
```diff
@@ -61,12 +61,24 @@ define([ "dialog/dialog", "util/lang", "ui/user-data", "ui/widget/tooltip" ],
...
-        _saveButton.classList.add( "btn-light" );
+      }
+    }
+
+    function togglePreviewButton( on ) {
+      if ( on ) {
+        _previewBtn.classList.remove( "butter-disabled" );
+        _previewBtn.href = butter.project.publishUrl;
+        _previewBtn.onclick = function() {
+          return true;
```
Comment (cadecairos) [inline src/ui/header.js:74]: same question as above

**Pull request closed without merging**